.github/workflows/publish.yml

```diff
@@ -25,7 +26,31 @@ jobs:
           echo "Using version: $VERSION"
           BRANCH_NAME="update-release-notes-js-$VERSION"
           echo "release_notes_branch=$BRANCH_NAME" >> "$GITHUB_OUTPUT"
+
+      - name: Check PR Status (Exists, Merged)
+        id: check_pr_status
+        run: |
+          BRANCH_NAME="${{ steps.determine-branch-name.outputs.release_notes_branch }}"
+          echo "Checking status for PR associated with branch: $BRANCH_NAME"
+
+          # Attempt to get the state. Let the command fail if PR not found.
+          # Check if the state is MERGED. Output 'true' or 'false'.
+          MERGED_OUTPUT=$(gh pr view "$BRANCH_NAME" --repo "${{ env.DOCS_REPO }}" --json state --jq '.state == "MERGED" | tostring')
+          EXIT_CODE=$?
```

Comment:
[q] Are you sure that if the previous command fails, the step won't fail immediately? You might need `set +e` at the beginning of `run`

Reply:
Yeah, you are right, added `set +e` now.